askup/utils/general.py

```diff
@@ -492,48 +464,41 @@ def get_student_last_week_incorrect_answers_count(user_id):
         )
         return cursor.fetchone()[0] or 0
 
-    return 0
-
 
-def get_user_organizations_queryset(user):
+def get_user_organizations_queryset(user_id, ordered=True):
     """
     Return the queryset of sorted by name user organizations.
     """
-    user = User.objects.get(pk=user) if isinstance(user, int) else user
-    return askup.models.Organization.objects.filter(users__in=[user]).order_by('name')
+    queryset = askup.models.Organization.objects.filter(users__in=[user_id])
 
+    if ordered:
+        queryset = queryset.order_by('name')
 
-def get_user_organizations_string(user):
-    """
-    Return comma separated user organizations string.
-    """
-    return ', '.join(
-        (str(organization) for organization in get_user_organizations_queryset(user))
-    )
+    return queryset
 
 
-def get_user_organizations_for_filter(user):
+def get_user_organizations_for_filter(user_id):
     """
     Return user organizations as a list of dictionaries in a sorted manner.
     """
-    return list(get_user_organizations_queryset(user).values('id', 'name'))
+    return list(get_user_organizations_queryset(user_id).values('id', 'name'))
 
 
-def get_first_user_organization(user):
+def get_first_user_organization(user_id):
     """
     Check if user is assigned to the specified organization.
     """
-    return get_user_organizations_queryset(user).first()
+    return get_user_organizations_queryset(user_id).first()
 
 
-def get_checked_user_organization_by_id(user, organization_id):
+def get_checked_user_organization_by_id(user_id, organization_id):
     """
     Check if an organization is assigned to the specified user and return it on success.
 
     Return an organization if it belongs to this user and None if it is not.
     """
-    user = User.objects.get(pk=user) if isinstance(user, int) else user
-    queryset = askup.models.Organization.objects.filter(users__in=[user], id=organization_id)
+    queryset = get_user_organizations_queryset(user_id, ordered=False)
```

Comment:
Why do we need `ordered=False` here?

Reply:
For not to make redundant index look ups on the PostgreSQL server site because we are searching only one organization that we need.

Reply:
In the case when we are interested only in one object I would add filter param to the function `get_user_organizations_queryset` instead of ordered